models/articles.models.js

```diff
@@ -39,18 +39,49 @@ exports.changeArticleById = (article_id, inc_votes) => {
   }
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.*, COUNT(comments.article_id)::INTEGER AS comment_count
+// db.query("SELECT ARRAY_AGG(topic) FROM articles ORDER BY topic");
+
+exports.fetchArticles = async (
+  sort_by = "created_at",
+  order = "desc",
+  topic
+) => {
+  const queryValues = [];
+  const Topics = await db.query("SELECT slug FROM topics;");
+  const validTopics = Topics.rows.map((topic) => topic.slug);
+  const validsort_by = [
+    "created_at",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "votes",
+  ];
+  const validorder = ["asc", "desc"];
+  let querystr = `SELECT articles.*, COUNT(comments.article_id)::INTEGER AS comment_count
     FROM comments
     LEFT JOIN articles
-    ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id;`
-    )
-    .then((result) => {
+    ON articles.article_id = comments.article_id`;
+  if (validTopics.includes(topic)) {
+    querystr += ` WHERE articles.topic = $1`;
+    queryValues.push(topic);
+  }
+  querystr += ` GROUP BY articles.article_id`;
+  if (validsort_by.includes(sort_by) && validorder.includes(order)) {
+    querystr += ` ORDER BY ${sort_by} ${order};`;
+  } else {
```

Comment:
Why not put this at the top of your file? Rejecting while building your query string seems like it would use too much processing power when you're working for Google...

Reply:
I already work for google i thought i told you already